olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -550,15 +551,23 @@ uint64_t DefaultCacheImpl::MaybeEvictData(leveldb::WriteBatch& batch) {
       continue;
     }
 
-    // Remove the key
-    batch.Delete(key);
-    evicted += key.size() + properties.size;
+    const auto expiry_key = CreateExpiryKey(key);
+    const auto current_eviction_size =
+        key.size() + properties.size + expiry_key.size() + kExpiryValueSize;
 
-    // Remove the key's expiry
-    auto expiry_key = CreateExpiryKey(key);
-    batch.Delete(expiry_key);
-    evicted += expiry_key.size() + kExpiryValueSize;
+    if (eviction_limit && evicted + current_eviction_size > eviction_limit) {
+      OLP_SDK_LOG_INFO_F(
```

Comment:
Rather debug. We only need a log at the end on how many items were evicted in total.